pkg/pdfcpu/types/streamdict.go

```diff
@@ -152,6 +237,11 @@ func parmsForFilter(d Dict) map[string]int {
 
 // Encode applies sd's filter pipeline to sd.Content in order to produce sd.Raw.
 func (sd *StreamDict) Encode() error {
+	if sd.Content == nil && sd.Raw != nil {
```

Comment:
(if Content == nil) then we don't have anything to encode regardless of sd.Raw. I would omit checking sd.Raw here also broadening this safeguard.

Reply:
Right now the check handles streams that have data which was not decoded, so there is no need to encode or update any properties.

If it's a "regular" stream with an empty `Content`, don't you have to at least update `sd.StreamLength` and the `Length` entry in the map?

Reply:
ok then let's do it your way.